src/filesystem_common.cpp

```diff
@@ -124,7 +124,12 @@ bool blacklist_pattern_list::match_dir(const std::string& name) const
 
 std::string get_prefs_file()
 {
-	return get_user_config_dir() + "/preferences";
```

Comment:
Why is this changed?

Reply:
Because the user config dir is not what will be synced.

Reply:
But the preferences file will continue to exist, right? You're just adding another preferences file in the sync dir?

Reply:
It doesn't make sense to me to have the synced and unsynced preferences saved to entirely different locations though.

Reply:
I can move the unsynced preferences back if you prefer.

Correct, syncing will not work. Steam expects the userdata to be in a specific place for syncing to work today as well - if someone moves it somewhere else, then their data will no longer be synced.

Reply:
There would also need to be some sort of handling for unknown preferences added as well, since at least on Linux the config dir tends to not be versioned so with the current implementation in this PR it would result in unsynced preferences added to the dev version being lost when someone switches between the stable and dev versions.

Reply:
1. I would keep `get_prefs_file()` as the function to return the normal preferences file.
2. I would either properly use the user config dir (leave the preferences there and put the sync dir there as well) or remove support for it.

Reply:
I don't know what you mean by "put the sync dir there as well" - only a single directory gets synced by Steam, it doesn't accomplish anything to potentially have a sync dir in two places. I will remove the config dir functionality in a separate PR though.

Reply:
The sync dir should be where the preference file is. The preference file is in the user config dir at the moment. So logically the sync dir would be in the user config dir. No two locations, just one.

Since you appear to not like the user config dir the other option is to remove support for it and move everything to the user data dir. A mix of both makes little sense.

Reply:
Why should the sync dir be where the preference file is? From your message, it's not clear what your reasoning is.

Reply:
This makes no sense at all. Currently what's synced is the saves folder, not the preferences folder, and it would be absolutely bizarre to move everything that is going to be synced to the user config directory instead.

Reply:
Sorry must have missed the notification for this.

Looks like I got confused that this is only about preferences. So it sounds like the only option is to remove the user config dir. If that is not supposed to happen in this PR then the preference file should not move in this PR either.

Reply:
Again, what's your reasoning? Why are those the two options?

Reply:
What other options are there? The preference file should move to the user data dir so it is next to the sync dir. It makes no sense to have a config dir with no config in it. No two options left. Just one.

Reply:
I intend to do this and figure out how to get the assert discussed in https://github.com/wesnoth/wesnoth/pull/9016#discussion_r1650036348 working in a followup PR. The preferences location also already defaults to the userdata dir as of #8848 anyway - all this PR does is split it into two files and and move one file into a subdirectory.

Reply:
As far as I can tell this PR currently changes `get_prefs_file` to return the path to the new synced preferences and adds the function `get_unsynced_prefs_file` to return the path to the preference file but from the user data dir.

I would keep `get_prefs_file` as is since there is no intention to remove the user config dir in this PR and add a `get_synced_prefs_file` to return the path to the new synced preferences file.

Reply:
I've renamed it to `get_synced_prefs_file`, that way it's clear what each is meant to do.